WPMedia/fusion-news-theme-blocks
# PEN-1745 Primary and Secondary Font Components

## Description

Utilising styled components to reduce the duplication of theme styles

Updated the headline block and numbered list block to show how it can be used.

## Testing

1. Run fusion with `npx fusion start -f -l @wpmedia/numbered-list-block,@wpmedia/headline-block,@wpmedia/shared-styles`
2. Goto an article page - http://localhost/arts/2020/01/17/a-huge-loss-for-lovers-of-romantic-music-chamin-correa-legendary-mexican-guitarist-dies-at-90/?_website=the-gazette
3. Verify article heading is using correct font
4. Goto an article page on a different site in the multi site set up - http://localhost/arts/2020/01/17/a-huge-loss-for-lovers-of-romantic-music-chamin-correa-legendary-mexican-guitarist-dies-at-90/?_website=the-sun
5. Verify article heading is used correct font for site
6. Also verify numbered-list-block 


## Jira Ticket
- [PEN-1745](https://arcpublishing.atlassian.net/browse/PEN-1745)


## Before

<img width="2705" alt="resuable-primary-font-block-before" src="https://user-images.githubusercontent.com/868127/106890645-89c50f80-66e1-11eb-80d1-63d1cf403a3b.png">


## After

<img width="3000" alt="resuable-primary-font-block-after" src="https://user-images.githubusercontent.com/868127/106890653-8c276980-66e1-11eb-9c86-4f9c2ad66c9b.png">



![Screen Shot 2021-02-03 at 17 16 10](https://user-images.githubusercontent.com/5950956/106822119-1de09980-6644-11eb-9e35-31bee2bd8fbd.png)
- this is a slight compromise to keep some of these more presentational components to not destructure the crazy fusion props 


todo: 

- [x] Test with multiple primary-font switching based off of different sites https://github.com/WPMedia/Fusion-News-Theme/blob/master/blocks.json#L158 vs https://github.com/WPMedia/Fusion-News-Theme/blob/master/blocks.json#L194
- [x] Get lighthouse score and/or unused css for before/after this pr -> todo: check after https://corecomponents.arcpublishing.com/pf/all-blocks/?_website=washpost
- [x] Perhaps find a couple other places in the blocks that this could be used to really scale this

## lighthouse/perf 

### before 

- key part is css in the html doc, and notably total bytes, rather than unused (they're used but just not necessarily different, duplicated)
![Screen Shot 2021-02-04 at 13 59 29](https://user-images.githubusercontent.com/5950956/106948416-4c687e00-66f1-11eb-8ded-ec957964dca8.png)

- reduce the amount of style calculations 
- https://developers.google.com/web/fundamentals/performance/rendering/reduce-the-scope-and-complexity-of-style-calculations?utm_source=lighthouse&utm_medium=devtools 
- 
![Screen Shot 2021-02-04 at 14 06 05](https://user-images.githubusercontent.com/5950956/106949109-2db6b700-66f2-11eb-8bd7-3a1763efe612.png)
![Screen Shot 2021-02-04 at 14 03 48](https://user-images.githubusercontent.com/5950956/106949118-30191100-66f2-11eb-95c6-64681d71a305.png)

### after

![Screen Shot 2021-02-04 at 15 03 21](https://user-images.githubusercontent.com/5950956/106954989-30b5a580-66fa-11eb-9991-1c3eb939fa7c.png)
